src/azure-cli-core/azure/cli/core/_help.py

```diff
@@ -117,15 +121,26 @@ def print_arguments(help_file):
 
     return indent
 
-def get_group_priority(group_name):
-    priorities = {
-        None: 0,
-        'Generic Update Arguments': 5,
-        'Resource Id Arguments': 10,
-        'Global Arguments': 1000,
-        }
-    key = priorities.get(group_name, 1)
-    return "%06d" % key
+class ArgumentGroupRegistry(object): # pylint: disable=too-few-public-methods
+
+    def __init__(self, group_list):
+
+        self.priorities = {
+            None: 0,
+            'Resource Id Arguments': 1,
+            'Generic Update Arguments': 998,
+            'Global Arguments': 1000,
+            }
+        priority = 2
+        # get alphabetical list of groups and ensure they are in the group priority dictionary
+        group_list = sorted(list(set(group_list)))
+        for group in [g for g in group_list if g not in self.priorities]:
+            self.priorities[group] = priority
+            priority += 1
+
+    def get_group_priority(self, group_name):
+        key = self.priorities.get(group_name, 0)
+        return "%06d" % key
```

Comment:
just a question, why we need to expand to 6 digits, is the comparison based on string, not int?